src/Moq/ActionObserver.cs

```diff
@@ -168,6 +256,18 @@ private Prettifier()
 			{
 			}
 
+			protected override Expression VisitMember(MemberExpression node)
+			{
+				if (node.Member is FieldInfo field && node.Expression is ConstantExpression constant)
+				{
+					return Expression.Constant(field.GetValue(constant.Value));
```

Comment:
Rebasing this PR on top of #768 has led to this being duplicated code: The same logic now already exists in `CapturedVariablesEvaluator`. Might want to check whether we can merge the two somehow.

Reply:
Done. Two expression tree rewriters have been moved into a dedicated namespace, I am anticipating that there will eventually be a few more of those.